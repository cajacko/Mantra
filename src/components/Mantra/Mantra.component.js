/* eslint max-lines: 0 */
import React, { Component } from 'react';
import { Alert, Animated } from 'react-native';
import PropTypes from 'prop-types';
import MantraRender from 'components/Mantra/Mantra.render';

const outAnimationTiming = 300;
const animationTimeout = 100;

/**
 * Mantra Component, handles logic, state and animations when rendering a
 * users mantra in a list
 *
 * @type {class}
 */
class Mantra extends Component {
  /**
   * Initialise class
   *
   * @param  {object} props Props passed to component
   * @return {void}       No return
   */
  constructor(props) {
    super(props);

    this.state = {
      height: null,
      deletedManually: false,
      offlineInit: !props.online,
      syncOpacity: new Animated.Value(1),
      syncing: props.syncing,
    };

    this.height = null;

    this.deleteMantra = this.deleteMantra.bind(this);
    this.animateIn = this.animateIn.bind(this);
    this.onLayout = this.onLayout.bind(this);
  }

  /**
   * Runs when component recieves new props. Use to start animating out if
   * deleted, start fading the sync icon if online and whether to rotate the
   * sync icon
   *
   * @param  {boolean} deletedProp  Is the mantra deletedProp
   * @param  {boolean} deletedState Was the mantra previosuly deleted
   * @param  {boolean} online       Is the mantra saved online
   * @param  {boolean} syncing      Is the mantra currently syncing
   * @return {void}              No return value
   */
  componentWillReceiveProps({ deletedProp, deletedState, online, syncing }) {
    if (this.state.deletedManually) {
      return;
    }

    if (deletedProp && deletedState === false) {
      this.animateOut();
    }

    if (online && !this.props.online) {
      this.hideSyncIcon();
    }

    if (this.state.syncing && !syncing && !online) {
      this.setState({ syncing: false });
    } else if (syncing && !this.state.syncing) {
      this.setState({ syncing });
    }
  }

  /**
   * Set the height of the mantra item when it is laid out. For use later when
   * animating the collapsing
   *
   * @param  {event} event The layout event object, used to get height
   * @return {void}       No return
   */
  onLayout(event) {
    this.height = event.nativeEvent.layout.height;

    if (this.height !== null && !this.props.initial) {
      this.animateIn();
    }
  }

  /**
   * Start animating the sync icons opacity to 0
   *
   * @return {void} No return value
   */
  hideSyncIcon() {
    Animated.timing(this.state.syncOpacity, {
      toValue: 0,
      duration: outAnimationTiming,
    }).start();
  }

  /**
   * Animate in a Mantra, whose height was not originally set. Animate heihgt
   * from 0 to where it should be
   *
   * @return {void} No return value
   */
  animateIn() {
    if (this.state.height === null) {
      const height = this.height;

      this.setState({
        height: new Animated.Value(0),
      });

      setTimeout(() => {
        Animated.timing(this.state.height, {
          toValue: height,
          duration: outAnimationTiming,
        }).start();
      }, animationTimeout);
    }
  }

  /**
   * Animate a Mantra's height from full to 0
   * @param  {FunctionDeclaration} callback What to do when the animation is
   * complete
   * @return {void}              No return value
   */
  animateOut(callback = () => {}) {
    this.setState({
      height: new Animated.Value(this.height),
    });

    setTimeout(() => {
      Animated.timing(this.state.height, {
        toValue: 0,
        duration: outAnimationTiming,
      }).start(callback);
    }, animationTimeout);
  }

  /**
   * Show an alert when deleting a Mantra item
   *
   * @return {void} No return value
   */
  deleteMantra() {
    Alert.alert('Delete Post', 'Are you sure you want to delete this post?', [
      {
        text: 'Delete',
        onPress: () => {
          this.animateOut(() => {
            this.setState({ deletedManually: true });
            this.props.deleteMantra();
          });
        },
      },
      { text: 'Back' },
    ]);
  }

  /**
   * Pass the props and state needed for the render component to do its thing
   *
   * @return {component} JSX component
   */
  render() {
    return (
      <MantraRender
        showSyncIcon={this.state.offlineInit}
        syncOpacity={this.state.syncOpacity}
        height={this.state.height}
        title={this.props.title}
        onLayout={this.onLayout}
        onPress={this.props.editMantra}
        syncing={this.state.syncing}
        rotation={this.props.rotation}
        initial={this.props.initial}
      />
    );
  }
}

Mantra.propTypes = {
  title: PropTypes.string.isRequired,
  deleteMantra: PropTypes.func.isRequired,
  online: PropTypes.bool.isRequired,
  editMantra: PropTypes.func.isRequired,
  syncing: PropTypes.bool.isRequired,
  // eslint-disable-next-line
  rotation: PropTypes.object,
  initial: PropTypes.bool.isRequired,
  deletedProp: PropTypes.bool.isRequired,
  deletedState: PropTypes.bool.isRequired,
};

Mantra.defaultProps = {
  rotation: null,
};

export default Mantra;
