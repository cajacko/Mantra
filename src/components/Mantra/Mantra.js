import React from 'react';
import { View, Text } from 'react-native';
import PropTypes from 'prop-types';
import { Ionicons } from '@expo/vector-icons';
import Swipeout from 'react-native-swipeout';
import style from 'components/Mantra/Mantra.style';

function swipeoutBtns(deleteAction, edit) {
  return [
    { text: 'Delete', onPress: deleteAction, backgroundColor: style.deleteBackground, color: style.buttonColour },
    { text: 'Edit', onPress: edit, backgroundColor: style.editBackground, color: style.buttonColour },
  ];
}

const Mantra = ({ title, last, deleteMantra, editMantra, online }) => {
  const containerStyles = [style.container];
  let syncIcon;

  if (last) {
    containerStyles.push(style.last);
  }

  if (!online) {
    syncIcon = (
      <View style={style.icon}>
        <Ionicons name="ios-sync" size={style.iconSize} color={style.iconColour} />
      </View>
    );
  }

  return (
    <Swipeout
      right={swipeoutBtns(deleteMantra, editMantra)}
      style={containerStyles}
      autoClose
      sensitivity={50}
    >
      <View style={style.wrapper}>
        <Text style={style.text}>{title}</Text>
        {syncIcon}
      </View>
    </Swipeout>
  );
};

Mantra.propTypes = {
  title: PropTypes.string.isRequired,
  last: PropTypes.bool.isRequired,
  deleteMantra: PropTypes.func.isRequired,
  online: PropTypes.bool.isRequired,
  editMantra: PropTypes.func.isRequired,
};

export default Mantra;
