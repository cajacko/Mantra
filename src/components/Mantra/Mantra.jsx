import React from 'react';
import style from 'components/Mantra/Mantra.style';

class MantraLoop extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      title: props.title,
      description: props.description,
    };

    this.onTitleChange = this.onTitleChange.bind(this);
    this.onDescriptionChange = this.onDescriptionChange.bind(this);
    this.saveMantra = this.saveMantra.bind(this);
  }

  onTitleChange(event) {
    this.setState({ title: event.target.value });
  }

  onDescriptionChange(event) {
    this.setState({ description: event.target.value });
  }

  saveMantra() {
    this.props.saveMantra(this.state.title, this.state.description);

    if (this.props.temp) {
      this.setState({
        title: '',
        description: '',
      });
    }
  }

  isEdited() {
    if (this.props.title !== this.state.title) {
      return true;
    }

    if (this.props.description !== this.state.description) {
      return true;
    }

    return false;
  }

  render() {
    let save;
    let deleteButton;

    if (this.isEdited()) {
      save = <button onClick={this.saveMantra}>Save</button>;
    }

    if (!this.props.temp) {
      deleteButton = <button onClick={this.props.deleteMantra}>Delete</button>;
    }

    return (
      <article style={style.container}>
        <input
          style={style.title}
          type="text"
          value={this.state.title}
          onChange={this.onTitleChange}
          placeholder="Add mantra title here"
        />
        <textarea
          style={style.description}
          value={this.state.description}
          onChange={this.onDescriptionChange}
          placeholder="Add mantra description here"
        />
        <footer style={style.footer}>
          {save}
          {deleteButton}
        </footer>
      </article>
    );
  }
}

export default MantraLoop;
