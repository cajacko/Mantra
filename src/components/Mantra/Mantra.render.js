/* eslint max-lines: 0 */
import React, { PureComponent } from 'react';
import { View, Text, Animated, TouchableOpacity } from 'react-native';
import PropTypes from 'prop-types';
// eslint-disable-next-line
import { Ionicons } from '@expo/vector-icons';
import style from 'components/Mantra/Mantra.style';
import IconButton from 'components/IconButton/IconButton.render';

/**
 * Mantra render component. Purely responsible to markup, styles and
 * if/else/loop logic. Nothing else
 * @type {class}
 */
class Mantra extends PureComponent {
  /**
   * Initialise the component and bind the methods
   *
   * @param  {Object} props Props passed ot the component
   * @return {void}       No return value
   */
  constructor(props) {
    super(props);

    this.syncStyles = this.syncStyles.bind(this);
  }

  /**
   * Logic to decide on the styles for the sync icon, with highlight and
   * rotation if syncing
   *
   * @return {Object} Object mapping the styles for the icon
   */
  syncStyles() {
    let iconColour;
    let syncStyle;

    if (this.props.showSyncIcon) {
      iconColour = style.iconColour;
      syncStyle = { ...style.icon, opacity: this.props.syncOpacity };

      if (this.props.syncing) {
        iconColour = style.iconColourSyncing;
        syncStyle = {
          ...syncStyle,
          transform: [{ rotate: this.props.rotation }],
        };
      }
    }

    return { iconColour, syncStyle };
  }

  /**
   * Render method to decide on markup and style
   * @return {jsx} Return jsx markup
   */
  render() {
    const { iconColour, syncStyle } = this.syncStyles();

    let heightStyle = {};

    if (this.props.height !== null) {
      heightStyle = { height: this.props.height };
    }

    let viewStyle = {};

    if (!this.props.initial && this.props.height === null) {
      viewStyle = {
        position: 'absolute',
        opacity: 0,
      };
    }

    const PressView = this.props.onPress ? TouchableOpacity : View;

    return (
      <View onLayout={this.props.onLayout} style={viewStyle}>
        <Animated.View style={{ overflow: 'hidden', ...heightStyle }}>
          <PressView onPress={this.props.onPress && this.props.onPress}>
            <View style={style.container}>
              {this.props.add && (
                <View style={style.add}>
                  <IconButton
                    icon="ios-add-outline"
                    onPress={this.props.add}
                    hasBackground
                  />
                </View>
              )}
              <View style={style.wrapper}>
                <Text style={style.text}>{this.props.title}</Text>
                {this.props.showSyncIcon && (
                  <Animated.View style={syncStyle}>
                    <Ionicons
                      name="ios-sync"
                      size={style.iconSize}
                      color={iconColour}
                    />
                  </Animated.View>
                )}
              </View>
              {this.props.discard && (
                <View style={style.discard}>
                  <IconButton
                    icon="ios-close-outline"
                    onPress={this.props.discard}
                    hasBackground={false}
                  />
                </View>
              )}
            </View>
          </PressView>
        </Animated.View>
      </View>
    );
  }
}

Mantra.propTypes = {
  // eslint-disable-next-line react/forbid-prop-types
  syncOpacity: PropTypes.object,
  showSyncIcon: PropTypes.bool,
  height: PropTypes.number,
  title: PropTypes.string.isRequired,
  onLayout: PropTypes.func,
  onPress: PropTypes.func,
  syncing: PropTypes.bool,
  // eslint-disable-next-line react/forbid-prop-types
  rotation: PropTypes.object,
  initial: PropTypes.bool,
  discard: PropTypes.func,
  add: PropTypes.func,
};

Mantra.defaultProps = {
  rotation: null,
  height: null,
  syncOpacity: null,
  showSyncIcon: false,
  onLayout: null,
  syncing: false,
  initial: false,
  onPress: null,
  add: null,
  discard: null,
};

export default Mantra;
