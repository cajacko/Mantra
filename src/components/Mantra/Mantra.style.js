import { TEXT_SIZES } from 'constants/text';
import { HORIZONTAL_VIEW_SPACING } from 'constants/spacing';
import { WHITE, GREY_LIGHT, GREY, GREY_LIGHTER, BLACK } from 'constants/colours';

const horizontalSpacing = HORIZONTAL_VIEW_SPACING;
const verticalSpacing = 20;
const iconSize = 20;

export default {
  container: {
    borderTopWidth: 1,
    borderColor: GREY_LIGHT,
    backgroundColor: WHITE,
  },

  wrapper: {
    paddingLeft: horizontalSpacing,
    paddingRight: horizontalSpacing,
    paddingTop: verticalSpacing,
    paddingBottom: verticalSpacing,
    flexDirection: 'row',
    alignItems: 'center',
  },

  text: {
    fontSize: TEXT_SIZES.MEDIUM,
    flex: 1,
  },

  last: {
    borderBottomWidth: 1,
  },

  icon: {
    height: iconSize,
    overflow: 'hidden',
  },

  iconSize: iconSize,
  iconColour: GREY,
  deleteBackground: GREY_LIGHT,
  editBackground: GREY_LIGHTER,
  buttonColour: BLACK,
};
