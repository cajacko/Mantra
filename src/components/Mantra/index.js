export { default } from 'components/Mantra/Mantra.container';
