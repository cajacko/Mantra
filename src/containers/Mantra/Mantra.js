import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Mantra from 'components/Mantra/Mantra';
import switchView from 'actions/switchView';
import deleteMantra from 'actions/deleteMantra';

class MantraContainer extends Component {
  constructor(props) {
    super(props);
    this.state = { deleted: props.deleted };
    this.deleteMantra = this.deleteMantra.bind(this);
    this.editMantra = this.editMantra.bind(this);
  }

  deleteMantra() {
    this.props.dispatch(deleteMantra(this.props.id));
    this.setState({ deleted: true });
  }

  finishedProgramaticDeleteAnimation() {
    this.setState({ deleted: true });
  }

  editMantra() {
    this.props.dispatch(switchView('AddView', {
      id: this.props.id,
      title: this.props.title,
    }));
  }

  render() {
    if (this.state.deleted) {
      return null;
    }

    let syncing = false;

    if (this.props.syncLoading) {
      if (this.props.offlineItemsSyncing.includes(this.props.id)) {
        syncing = true;
      }
    }

    return (
      <Mantra
        deleteMantra={this.deleteMantra}
        title={this.props.title}
        last={this.props.last}
        online={this.props.online}
        editMantra={this.editMantra}
        rotation={this.props.rotation}
        syncing={syncing}
        initial={this.props.initial}
        deletedProp={this.props.deleted}
        deletedState={this.state.deleted}
      />
    );
  }
}

MantraContainer.propTypes = {
  dispatch: PropTypes.func.isRequired,
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  last: PropTypes.bool,
  online: PropTypes.bool.isRequired,
  // eslint-disable-next-line
  rotation: PropTypes.object,
  syncLoading: PropTypes.bool.isRequired,
  initial: PropTypes.bool.isRequired,
  deleted: PropTypes.bool.isRequired,
  offlineItemsSyncing: PropTypes.arrayOf(PropTypes.string).isRequired,
};

MantraContainer.defaultProps = {
  last: false,
  rotation: null,
};

function mapStateToProps({ syncLoading, offlineItemsSyncing }) {
  return { syncLoading, offlineItemsSyncing };
}

export default connect(mapStateToProps)(MantraContainer);
