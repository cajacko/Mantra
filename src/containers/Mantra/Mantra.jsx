import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Mantra from 'components/Mantra/Mantra';
import { saveMantra, deleteMantra } from 'actions/mantra';

class MantraLoopContainer extends React.Component {
  constructor(props) {
    super(props);

    this.saveMantra = this.saveMantra.bind(this);
    this.deleteMantra = this.deleteMantra.bind(this);
  }

  saveMantra(title, description) {
    this.props.dispatch(saveMantra(this.props.id, title, description));
  }

  deleteMantra() {
    this.props.dispatch(deleteMantra(this.props.id));
  }

  render() {
    return (
      <Mantra
        temp={this.props.temp}
        title={this.props.title}
        description={this.props.description}
        saveMantra={this.saveMantra}
        deleteMantra={this.deleteMantra}
      />
    );
  }
}

function mapStateToProps(state, { id }) {
  let title = 'Add new mantra here';
  let description = 'Add new mantra description here';
  const mantra = state.mantra.mantra[id];

  if (mantra) {
    title = mantra.title;
    description = mantra.description;
  }

  return { title, description };
}

export default connect(mapStateToProps)(MantraLoopContainer);
