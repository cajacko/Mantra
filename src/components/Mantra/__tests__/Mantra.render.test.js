import 'react-native';
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Mantra from '../Mantra.render';

describe('Mantra.render', () => {
  test('Renders correctly with default params', () => {
    const props = {
      title: 'Hello I am Mantra',
      initial: false,
      onLayout: () => {},
      rotation: { AnimatedMockRotation: true },
      onPress: () => {},
    };

    const renderer = new ShallowRenderer();
    renderer.render(<Mantra {...props} />);
    const result = renderer.getRenderOutput();

    expect(result).toMatchSnapshot();
  });

  test('With a specific height', () => {
    const props = {
      title: 'Hello I am Mantra',
      height: 200,
      initial: false,
      rotation: { AnimatedMockRotation: true },
      onPress: () => {},
    };

    const renderer = new ShallowRenderer();
    renderer.render(<Mantra {...props} />);
    const result = renderer.getRenderOutput();

    expect(result).toMatchSnapshot();
  });

  test('Initial and no height still shows', () => {
    const props = {
      title: 'Hello I am Mantra',
      height: null,
      initial: true,
      rotation: { AnimatedMockRotation: true },
      onPress: () => {},
    };

    const renderer = new ShallowRenderer();
    renderer.render(<Mantra {...props} />);
    const result = renderer.getRenderOutput();

    expect(result).toMatchSnapshot();
  });

  test('Show default sync icon', () => {
    const props = {
      title: 'Hello I am Mantra',
      height: null,
      initial: true,
      showSyncIcon: true,
      syncOpacity: { AnimatedMockOpacity: true },
      syncing: false,
      rotation: { AnimatedMockRotation: true },
      onPress: () => {},
    };

    const renderer = new ShallowRenderer();
    renderer.render(<Mantra {...props} />);
    const result = renderer.getRenderOutput();

    expect(result).toMatchSnapshot();
  });

  test('Show sync icon syncing', () => {
    const props = {
      title: 'Hello I am Mantra',
      height: null,
      initial: true,
      showSyncIcon: true,
      syncOpacity: { AnimatedMockOpacity: true },
      syncing: true,
      rotation: { AnimatedMockRotation: true },
      onPress: () => {},
    };

    const renderer = new ShallowRenderer();
    renderer.render(<Mantra {...props} />);
    const result = renderer.getRenderOutput();

    expect(result).toMatchSnapshot();
  });
});
