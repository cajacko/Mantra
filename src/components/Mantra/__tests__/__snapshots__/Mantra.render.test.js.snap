// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mantra.render Initial and no height still shows 1`] = `
<View
  onLayout={null}
  style={Object {}}
>
  <AnimatedComponent
    style={
      Object {
        "overflow": "hidden",
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderBottomWidth": 1,
            "borderColor": "#e2e2e2",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingBottom": 20,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 20,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "flex": 1,
                "fontSize": 20,
              }
            }
          >
            Hello I am Mantra
          </Text>
        </View>
      </View>
    </TouchableOpacity>
  </AnimatedComponent>
</View>
`;

exports[`Mantra.render Renders correctly with default params 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "opacity": 0,
      "position": "absolute",
    }
  }
>
  <AnimatedComponent
    style={
      Object {
        "overflow": "hidden",
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderBottomWidth": 1,
            "borderColor": "#e2e2e2",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingBottom": 20,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 20,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "flex": 1,
                "fontSize": 20,
              }
            }
          >
            Hello I am Mantra
          </Text>
        </View>
      </View>
    </TouchableOpacity>
  </AnimatedComponent>
</View>
`;

exports[`Mantra.render Show default sync icon 1`] = `
<View
  onLayout={null}
  style={Object {}}
>
  <AnimatedComponent
    style={
      Object {
        "overflow": "hidden",
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderBottomWidth": 1,
            "borderColor": "#e2e2e2",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingBottom": 20,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 20,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "flex": 1,
                "fontSize": 20,
              }
            }
          >
            Hello I am Mantra
          </Text>
          <AnimatedComponent
            style={
              Object {
                "height": 20,
                "opacity": Object {
                  "AnimatedMockOpacity": true,
                },
                "overflow": "hidden",
              }
            }
          >
            <Icon
              allowFontScaling={false}
              color="#828282"
              name="ios-sync"
              size={20}
            />
          </AnimatedComponent>
        </View>
      </View>
    </TouchableOpacity>
  </AnimatedComponent>
</View>
`;

exports[`Mantra.render Show sync icon syncing 1`] = `
<View
  onLayout={null}
  style={Object {}}
>
  <AnimatedComponent
    style={
      Object {
        "overflow": "hidden",
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderBottomWidth": 1,
            "borderColor": "#e2e2e2",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingBottom": 20,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 20,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "flex": 1,
                "fontSize": 20,
              }
            }
          >
            Hello I am Mantra
          </Text>
          <AnimatedComponent
            style={
              Object {
                "height": 20,
                "opacity": Object {
                  "AnimatedMockOpacity": true,
                },
                "overflow": "hidden",
                "transform": Array [
                  Object {
                    "rotate": Object {
                      "AnimatedMockRotation": true,
                    },
                  },
                ],
              }
            }
          >
            <Icon
              allowFontScaling={false}
              color="#3596c6"
              name="ios-sync"
              size={20}
            />
          </AnimatedComponent>
        </View>
      </View>
    </TouchableOpacity>
  </AnimatedComponent>
</View>
`;

exports[`Mantra.render With a specific height 1`] = `
<View
  onLayout={null}
  style={Object {}}
>
  <AnimatedComponent
    style={
      Object {
        "height": 200,
        "overflow": "hidden",
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderBottomWidth": 1,
            "borderColor": "#e2e2e2",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingBottom": 20,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 20,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "flex": 1,
                "fontSize": 20,
              }
            }
          >
            Hello I am Mantra
          </Text>
        </View>
      </View>
    </TouchableOpacity>
  </AnimatedComponent>
</View>
`;
